src/analysis/mean_inequalities.lean

```diff
@@ -377,11 +396,22 @@ begin
   simp only [← abs_mul, le_abs_self]
 end
 
+/-- For `1 ≤ p`, the `p`-th power of the sum of `f i` is bounded above by a constant times the
+sum of the `p`-th powers of `f i`. Version for sums over finite sets, with `ℝ`-valued functions. -/
+theorem rpow_sum_le_const_mul_sum_rpow (hp : 1 ≤ p) :
+  (∑ i in s, |f i|) ^ p ≤ (card s) ^ (p - 1) * ∑ i in s, |f i| ^ p :=
+begin
+  have := nnreal.coe_le_coe.2
+    (nnreal.rpow_sum_le_const_mul_sum_rpow s (λ i, ⟨_, abs_nonneg (f i)⟩) hp),
+  push_cast at this,
+  exact this, -- for some reason `exact_mod_cast` can't replace this argument
```

Comment:
This is cutting against the grain of `exact_mod_cast`. `norm_cast` wants to eliminate coercions, so it wants to make your `have` lemma about `nnreal`. You want it about `real` with the `nnreal`s inside cast up. So I think the way you've done it is right.